### PR TITLE
Enable getting model predictions using only the proba and the model_classes

### LINKED ISSUE #2090: [TABULAR] Design a way to infer the correct predicted class from probas
**Is your feature request related to a problem? Please describe.**
Currently in the tabular package predictions must be given / implemented, while predict_proba is optional. Theoretically (and done in practice in Vision) it's possible to get the predicted class from .argmax() on the proba. In tabular though, it's more complicated because classes don't have to be [0, .., n-1].

**Describe the solution you'd like**
Two possible solutions:
1. Leave as is, tabular users must give predictions
2. Use the context.classes to get the classes that correspond to the proba columns. Downside is that it practically assumes that all classes are in the label.

### PR DESCRIPTION
**Pull request by noamzbr**
Closes #2090

### CODE REVIEW

File: deepchecks/tabular/context.py [hunk 1]
```diff
@@ -88,7 +90,9 @@ def __init__(self,
             if dataset is not None:
                 feature_df_list.append(dataset.features_columns)
                 if y_pred is None and y_proba is not None:
+                    validate_proba(y_proba, model_classes)
                     y_pred = np.argmax(y_proba, axis=-1)
+                    y_pred = np.array(model_classes)[y_pred]
```
Comment (Nadav-Barak) [inline deepchecks/tabular/context.py:95]: Nice

File: tests/tabular/test_dummy_model.py [hunk 2]
```diff
@@ -220,3 +220,12 @@ def test_suite(diabetes_split_dataset_and_model):
     result = suite.run(**args)
     length = get_expected_results_length(suite, args)
     validate_suite_result(result, length)
+
+
+def test_predict_using_proba(iris_binary_string_split_dataset_and_model):
```
Comment (Nadav-Barak) [inline tests/tabular/test_dummy_model.py:225]: can compare the check result when running on probas vs running on predictions

**Review by Nadav-Barak — APPROVED** [review]

File: deepchecks/tabular/context.py (hunk 1 above)
Comment (matanper) [reply deepchecks/tabular/context.py:95]: also possible `np.argmax(y_proba, axis=-1).map(model_classes.__getitem__)`

**Review by matanper — APPROVED** [review]

File: deepchecks/tabular/context.py (hunk 1 above)
Comment (noamzbr) [reply deepchecks/tabular/context.py:95]: Mine is more elegant

**Pull request merged**